lib/class-wp-rest-menu-items-controller.php

```diff
@@ -181,19 +203,35 @@ public function create_item( $request ) {
 	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
 	 */
 	public function update_item( $request ) {
-		$valid_check = $this->get_nav_menu_item( $request['id'] );
+		$prepared_nav_item = $this->update_item_validate( $request );
+		if ( is_wp_error( $prepared_nav_item ) ) {
+			return $prepared_nav_item;
+		}
+		$nav_menu_item = $this->update_item_persist( $prepared_nav_item, $request, $request );
+		$this->update_item_notify( $nav_menu_item, $request );
+
+		$response = $this->prepare_item_for_response( $nav_menu_item, $request );
+
+		return rest_ensure_response( $response );
+	}
+
+	public function update_item_validate( $input ) {
+		$valid_check = $this->get_nav_menu_item( $input['id'] );
 		if ( is_wp_error( $valid_check ) ) {
 			return $valid_check;
 		}
 
-		$prepared_nav_item = $this->prepare_item_for_database( $request );
+		$prepared_nav_item = $this->prepare_item_for_database( $input );
 
 		if ( is_wp_error( $prepared_nav_item ) ) {
 			return $prepared_nav_item;
 		}
 
 		$prepared_nav_item = (array) $prepared_nav_item;
+		return $prepared_nav_item;
+	}
 
+	public function update_item_persist( $prepared_nav_item, $input, $request ) {
```

Comment:
Why do we need to add the new `$input` parameter to this controller's method? Can't we stick to `$request`?

Reply:
When this is used by the batch processor, `$request` contains the information about the entire navigation and not about just one menu item. I used `$input` to pass the menu item information instead.